tests/rkt_privatenet_test.go

```diff
@@ -0,0 +1,325 @@
+
+	var wg sync.WaitGroup
+
+	// Host opens the server
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := test_netutils.HttpServe(httpServeAddr, httpServeTimeout)
+		if err != nil {
+			t.Fatalf("Error during HttpServe: %v", err)
+		}
+	}()
+
+	// Child connects to host
+	wg.Add(1)
+	hostname, err := os.Hostname()
+	go func() {
+		defer wg.Done()
+		cmd := fmt.Sprintf("%s --debug --insecure-skip-verify run --private-net=default --mds-register=false %s", ctx.cmd(), testImage)
+		t.Logf("Command: %v", cmd)
+		child, err := gexpect.Spawn(cmd)
+		if err != nil {
+			t.Fatalf("Cannot exec rkt: %v", err)
+		}
+		expectedRegex := `HTTP-Get received: (.*)\r`
+		result, _ := child.ExpectRegexFind(expectedRegex)
+		if len(result) == 0 {
+			t.Fatalf("Expected %q but not found", expectedRegex)
+		}
+		if result[1] != hostname {
+			t.Fatalf("Hostname received by client `%v` doesn't match `%v`", result[1], hostname)
+		}
+
+		err = child.Wait()
+		if err != nil {
+			t.Fatalf("rkt didn't terminate correctly: %v", err)
+		}
+	}()
+
+	wg.Wait()
+
+}
+
+/*
+ * Default-restricted private-net
+ * ---
+ * Container launches http server on all its interfaces
+ * Host must be able to connects to container's http server via container's
+ * eth0's IPv4
+ * TODO: verify that the container isn't NATed
+ */
+func TestPrivateNetDefaultRestrictedConnectivity(t *testing.T) {
+	httpServeAddr := "0.0.0.0:54321"
+	iface := "eth0"
+
+	testImageArgs := []string{fmt.Sprintf("--exec=/inspect --print-ipv4=%v --serve-http=%v", iface, httpServeAddr)}
+	testImage := patchTestACI("rkt-inspect-networking.aci", testImageArgs...)
+	defer os.Remove(testImage)
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+	defer ctx.reset()
+
+	cmd := fmt.Sprintf("%s --debug --insecure-skip-verify run --private-net=default-restricted --mds-register=false %s", ctx.cmd(), testImage)
+	t.Logf("Command: %v", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec rkt: %v", err)
+	}
+
+	expectedRegex := `IPv4: (.*)\r`
+	result, _ := child.ExpectRegexFind(expectedRegex)
+	if len(result) == 0 {
+		t.Fatalf("Expected %q but not found", expectedRegex)
+	}
+	httpGetAddr := fmt.Sprintf("http://%v:54321", result[1])
+
+	var wg sync.WaitGroup
+
+	// Child opens the server
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err = child.Wait()
+		if err != nil {
+			t.Fatalf("rkt didn't terminate correctly: %v", err)
+		}
+	}()
+
+	// Host connects to the child
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		expectedRegex := `serving on`
+		result, _ := child.ExpectRegexFind(expectedRegex)
+		if len(result) == 0 {
+			t.Fatalf("Expected %q but not found", expectedRegex)
+		}
+		body, err := test_netutils.HttpGet(httpGetAddr)
+		if err != nil {
+			log.Fatalf("%v\n", err)
+		}
+		log.Printf("HTTP-Get received: %s", body)
+		if err != nil {
+			log.Fatalf("%v\n", err)
+		}
+	}()
+
+	wg.Wait()
+}
+
+/*
+ * Network Template for generating configs on the fly
+ */
+var networkTemplate = `
```

Comment:
Yes, I plan on writing out network config files files at runtime from this template.